tests/kani/Static/pub_static.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// kani-flags: --enable-unstable --function harness
+//! This test covers an issue we had with our public-fns implementation.
+//! We were not checking if a root was a function in the first place.
+//! https://github.com/model-checking/kani/issues/2047
+
+pub static DAYS_OF_WEEK: [char; 7] = ['s', 'm', 't', 'w', 't', 'f', 's'];
+
+#[no_mangle]
```

Comment:
Why not `kani::proof` (and `--harness`)?

Reply:
Because the bug this change fixes was on how we were collecting starting points for the public functions reachability mode. For the harnesses reachability mode, the predicate only passes for functions.